src/components/DataTable.tsx

```diff
@@ -107,6 +123,10 @@ export function DataTable<T extends object>(props: Props<T>) {
 
   const mainCheckbox = useRef<HTMLInputElement>(null);
 
+  const handleVerticalOverflowChange = (hasOverflow: boolean) => {
+    setHasVerticalOverflow(hasOverflow);
+  };
+
```

Comment:
This doesn't need to be a method. I checked and we call it only once. Even if it had multiple references, it'd need better co-location or useCallback.

Nonetheless, we can just directly call `setHasVerticalOverflow` unless I'm missing something?

Reply:
Yeah, you are right, we call it just once so we can call it while passing property to Table component.

```
<Table
        onVerticalOverflowChange={(hasOverflow) =>
          setHasVerticalOverflow(hasOverflow)
        }
      >
```